### DISCUSSION TITLE
log_to_metric field not used?

### DISCUSSION THREAD

**Discussion by syntastical** [vectordotdev/vector#20363] [General]
Am I crazy or is the only purpose of the field attribute to check if the attribute exists and is not null?  Feels like I gotta be missing something.  In the docs it says that if the value of the field is null then it will not emit a metric, but I've found that it throws an error when the value is null.  Can someone tell me how this is supposed to work?

**Comment by jszwedko** [comment]
Hi @syntastical !

The `field` is what is used for the value of the metric. This example for a gauge here might make it clearer: https://vector.dev/docs/reference/configuration/transforms/log_to_metric/#examples-gauges

**Reply by syntastical** [reply]
I see, thanks for the clarification.  I am using log_to_metric with two counter metrics with increment_by_value set to false.  I was thinking I could just set field to null when I didn't want to increment either of the metrics, but it short circuits when either of them are null or missing.  Looks like I'll probably have to make separate filters and feed their own log_to_metric.
